hphp/hack/src/hh_codesynthesis/agentGraphGenerator.py

```diff
@@ -154,6 +154,21 @@ def add_product_agent(self, agent_number: int) -> None:
     def add_edge(self, left_agent: int, right_agent: int) -> None:
         self.edges[left_agent].add(right_agent)
 
+    def validate_range_in_profile(self, degree: List[int], direction: str) -> bool:
+        for node, degree in enumerate(degree):
+            # Select a right profile to use.
+            agent_profile = self.solving_context.infra_agent_profile
```

Comment:
This would be better written with if else, I think, and would be a lot clearer what's happening

Reply:
Also added an exception handling.

Reply:
Also added an exception handling.